mesonbuild/interpreter/interpreter.py

```diff
@@ -192,7 +193,7 @@ def dump_value(self, arr, list_sep, indent):
 )
 
 TEST_KWARGS: T.List[KwargInfo] = [
-    KwargInfo('args', ContainerTypeInfo(list, (str, mesonlib.File, build.Target)),
+    KwargInfo('args', ContainerTypeInfo(list, (str, mesonlib.File, build.Target, build.CustomTargetIndex)),
```

Comment:
I don't think it's your bug but mine, but this should *not* allow `Target`, ti should be be `BuildTarget | CustomTarget`, otherwise a `RunTarget` would be allowed, which we don't want.

Reply:
Good catch; I think it should be fixed on top, possibly by creating a superclass of BuildTarget/CustomTarget. In fact, Target has methods such as `get_outputs()` that make very little sense for a RunTarget, and probably most of the type annotations that say "Target" actually mean `T.Union[BuildTarget, CustomTarget]`.

Reply:
Ok, I'll take this&mdash;there's possibly another instance for `add_install_script`, I have to write tests and then I'll send a separate pull request.